components/google_drive/actions/download-file/download-file.js

```diff
@@ -0,0 +1,91 @@
+const googleDrive = require("../../google_drive.app");
+const fs = require("fs");
+const stream = require("stream");
+const { promisify } = require("util");
+const { GOOGLE_DRIVE_MIME_TYPE_PREFIX } = require("../../constants");
+const googleWorkspaceExportFormats = require("../google-workspace-export-formats");
+
+/**
+ * Uses Google Drive API to download files to a `filePath` in the /tmp
+ * directory.
+ *
+ * Use `files.export` for Google Workspace files types (e.g.,
+ * `application/vnd.google-apps.document`) and `files.get` for other file types,
+ * as per the [Download files API guide](https://bit.ly/2ZbJvcn).
+ */
+module.exports = {
+  key: "google_drive-download-file",
+  name: "Download File",
+  description: "Download a file",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    googleDrive,
+    drive: {
+      propDefinition: [
+        googleDrive,
+        "watchedDrive",
+      ],
+      description:
+        "The drive to use. If not specified, your personal Google Drive will be used. If you are connected with any [Google Shared Drives](https://support.google.com/a/users/answer/9310351), you can select it here.",
+      optional: true,
+    },
+    fileId: {
+      propDefinition: [
+        googleDrive,
+        "fileId",
+        (c) => ({
+          drive: c.drive,
+        }),
+      ],
+      description: "The file to download",
+    },
+    filePath: {
+      type: "string",
+      label: "Destination File Path",
+      description:
+        "The destination path for the file in /tmp (e.g., `/tmp/myFile.csv`).",
+    },
+    mimeType: {
```

Comment:
Do you have any reason to use the `mimeType` from `app`? Looks like you are changing everything. I am wondering if you need to use this `propDefinition` here